diff --git a/include/SDL_config_iphoneos.h b/include/SDL_config_iphoneos.h
index a3bf2e8ff..d4cea8caf 100644
--- a/include/SDL_config_iphoneos.h
+++ b/include/SDL_config_iphoneos.h
@@ -167,19 +167,19 @@
 #define SDL_VIDEO_RENDER_OGL_ES2    1
 
 /* Metal supported on 64-bit devices running iOS 8.0 and tvOS 9.0 and newer */
-#if !TARGET_OS_SIMULATOR && !TARGET_CPU_ARM && ((__IPHONE_OS_VERSION_MIN_REQUIRED >= 80000) || (__TV_OS_VERSION_MIN_REQUIRED >= 90000))
-#define SDL_PLATFORM_SUPPORTS_METAL	1
-#else
-#define SDL_PLATFORM_SUPPORTS_METAL	0
-#endif
-
-#if SDL_PLATFORM_SUPPORTS_METAL
-#define SDL_VIDEO_RENDER_METAL  1
-#endif
-
-#if SDL_PLATFORM_SUPPORTS_METAL
-#define SDL_VIDEO_VULKAN 1
-#endif
+// #if !TARGET_OS_SIMULATOR && !TARGET_CPU_ARM && ((__IPHONE_OS_VERSION_MIN_REQUIRED >= 80000) || (__TV_OS_VERSION_MIN_REQUIRED >= 90000))
+// #define SDL_PLATFORM_SUPPORTS_METAL	1
+// #else
+// #define SDL_PLATFORM_SUPPORTS_METAL	0
+// #endif
+// 
+// #if SDL_PLATFORM_SUPPORTS_METAL
+// #define SDL_VIDEO_RENDER_METAL  1
+// #endif
+// 
+// #if SDL_PLATFORM_SUPPORTS_METAL
+// #define SDL_VIDEO_VULKAN 1
+// #endif
 
 /* Enable system power support */
 #define SDL_POWER_UIKIT 1
